### PR TITLE
use mint for bridge token import

### LINKED ISSUE #358: Use mint for token imported from bridge
it is better to mint tokens instead of transfer from bridge pallet according to the discussion.

### PR DESCRIPTION
**Pull request by suinuj**
The PR resolves #358, the bridge will use mint for the imported token.

### CODE REVIEW

**Review by wangminqi — APPROVED** [review]

File: pallets/bridge-transfer/src/lib.rs [hunk 1]
```diff
@@ -93,27 +97,20 @@ pub mod pallet {
...
 		) -> DispatchResult {
-			let source = T::BridgeOrigin::ensure_origin(origin)?;
-			// transfer to bridge account from external accounts is not allowed.
-			if source == to {
-				fail!(Error::<T>::InvalidCommand);
-			}
+			T::BridgeOrigin::ensure_origin(origin)?;
 
+			let total_issuance = <T as Config>::Currency::total_issuance();
+			if total_issuance + amount > T::MaximumIssuance::get() {
```
Comment (Kailai-Wang) [inline pallets/bridge-transfer/src/lib.rs]: do we need to check overflow here? Like `checked_add`? For example the `issue()` fn in pallet-balances: https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L1453

Or can we remove this check and rely on the error handling of `mint_into()`?

File: pallets/bridge-transfer/src/tests.rs [hunk 2]
```diff
@@ -59,35 +59,30 @@ fn transfer() {
...
+		assert_events(vec![Event::Balances(balances::Event::Deposit {
+			who: RELAYER_A,
 			amount: 10,
 		})]);
 	})
 }
 
 #[test]
-fn transfer_to_holdingaccount() {
+fn exceed_max_supply() {
```
Comment (Kailai-Wang) [inline pallets/bridge-transfer/src/tests.rs:91]: In this test we only checked the scenario where we transfer one-off balance that exceeds the limit, shall we have another test that verifies the behaviour when we have accumulated balance that exceeds the limit? E.g. the sum of 5 transfers exceeds the max supply.

File: pallets/bridge-transfer/src/lib.rs (hunk 1 above)
Comment (suinuj) [reply pallets/bridge-transfer/src/lib.rs]: agree, overflow check is needed.

File: pallets/bridge-transfer/src/tests.rs (hunk 2 above)
Comment (suinuj) [reply pallets/bridge-transfer/src/tests.rs:91]: will add this case.

**Review by Kailai-Wang — APPROVED** [review]
`MaximumIssuance` can only be updated via runtime upgrade, right? (In case we want to change that)

**Pull request merged**